src/base-immutable/base-immutable.ts

```diff
@@ -43,8 +43,8 @@ export const PropertyType = {
   ARRAY: 'array' as PropertyType,
 };
 
-export interface Property {
-  name: string;
+export interface Property<T = string> {
+  name: T;
```

Comment:
what is the point of this?

Reply:
To allow for stricter typing (i.e. `T` should be `keyof ValueType` but I'm having trouble accessing instance types from static properties — it's a known TypeScript limitations and I'm trying to find a better way)...